templates/todo/common/infra/bicep/app/apim-api.bicep

```diff
@@ -23,7 +23,7 @@ param webFrontendUrl string
 @description('Absolute URL of the backend service implementing this API.')
 param apiBackendUrl string
 
-var apiPolicyContent = replace(loadTextContent('../../../../../common/infra/bicep/core/gateway/apim-api-policy.xml'), '{0}', webFrontendUrl)
+var apiPolicyContent = replace(loadTextContent('../../../../../common/infra/shared/gateway/apim/apim-api-policy.xml'), '{0}', webFrontendUrl)
```

Comment:
I would add a comment in apim-api-policy.xml to reference this code as where the magic is happening :)  

When I saw this:
![image](https://user-images.githubusercontent.com/24213737/211476069-d76e89ef-6cab-4dd4-813d-ac5d54e2635d.png)

I was really confused about what that `{0}` meant for the origin.

Also, by setting the allowed origin to the webFrontendUrl only, I am guessing that debugging the web locally (on localhost) and using the API from Azure (not locally) will fail, as http://localhost is not an allowed origin.

Did we test that use case?

Reply:
I don't think this is any different than our current behavior without APIM deployed: https://github.com/Azure/azure-dev/issues/921. The user needs to modify the API's CORs settings if they want to debug a remote API.

Reply:
We can make a template change to allow all CORS origins in the future (local dev mode), but let's make it as a separate change.